Orange/widgets/visualize/owmosaic.py

```diff
@@ -461,6 +462,9 @@ def set_subset_data(self, data):
         except:
             self.subset_data = None
             self.Warning.incompatible_subset(shown=data is not None)
+        finally:
+            indices = list(set(e.id for e in self.subset_data))
```

Comment:
I would set `self.subset_indices` to `None` if `self.data` is `None`. Alternatively, this twisted code (not by your fault!) could be improved. I think that `set_subset_data` should be only:

    @Inputs.data_subset
    def set_subset_data(self, data):
        self.subset_data = data

The rest of this function belongs to `handleNewSignals` (or a separate method called by `handleNewSignals`). This is precisely what `handleNewSignals` is meant for. (It's possible that this widget was originally written before we had `handleNewSignals`.)